plugins/pixart-rf/fu-pxi-ble-device.c

```diff
@@ -240,25 +304,57 @@ fu_pxi_ble_device_check_support_report_id(FuPxiBleDevice *self, GError **error)
 		return FALSE;
 
 	rpt_desc.size = desc_size;
+
+	/* Get Report Descriptor */
 	if (!fu_udev_device_ioctl(FU_UDEV_DEVICE(self),
 				  HIDIOCGRDESC,
 				  (guint8 *)&rpt_desc,
 				  NULL,
 				  FU_PXI_DEVICE_IOCTL_TIMEOUT,
 				  error))
 		return FALSE;
+
 	fu_dump_raw(G_LOG_DOMAIN, "HID descriptor", rpt_desc.value, rpt_desc.size);
 
 	/* check ota retransmit feature report usage page exist or not */
 	fu_byte_array_append_uint16(req, PXI_HID_DEV_OTA_RETRANSMIT_USAGE_PAGE, G_LITTLE_ENDIAN);
-	if (!fu_pxi_ble_device_search_hid_usage_page(rpt_desc.value,
-						     rpt_desc.size,
-						     req->data,
-						     req->len)) {
-		/* replace retransmit report id with feature report id, if retransmit report id not
-		 * found */
-		self->retransmit_id = PXI_HID_DEV_OTA_FEATURE_REPORT_ID;
+	if (!fu_pxi_ble_device_search_hid_feature_report_id(rpt_desc.value,
+					rpt_desc.size,
+					PXI_HID_DEV_OTA_RETRANSMIT_USAGE_PAGE,
+					&report_id,
+					error)) {
+		self->retransmit_id = report_id;
+	} else {
+
+		self->retransmit_id = PXI_HID_DEV_OTA_RETRANSMIT_REPORT_ID;
 	}
+
```

Comment:
the error handling here is funky. I'd have something like `g_autoptr(GError) error_local = NULL;` and then use `&error_local` in `fu_pxi_ble_device_search_hid_feature_report_id()` -- and on failure, do `g_debug("failed to parse descriptor: %s", error_local->message);` and fallback to `self->retransmit_id = PXI_HID_DEV_OTA_RETRANSMIT_REPORT_ID;`